### ISSUE TITLE
Transient issues during parallel rspec

### ISSUE THREAD

**Issue by viraptor** [Shopify/bootsnap#110]
I'm seeing a number of failures during parallel rspec runs. (~ 1 in 3000 specs) It looks like this:

```
  | Failure/Error:
  | ShoppingCart::AddedConfirmationPresenter.new(cart: cart,
  | item: item,
  | cart_entry: cart_entry,
  | keep_browsing_path: keep_browsing_path)
  |  
  | LoadError:
  | cannot load such file -- /var/lib/buildkite-agent/builds/redacted/envato-marketplaces/marketplace/app/presenters/shopping_cart/added_confirmation_presenter
  | Shared Example Group: "a guide structure and a real view model" called from ./spec/documentation/guide/content/structures/purchase_flow/shopping_cart/added_confirmation_modal_spec.rb:18
  | # ./vendor/bundle/ruby/2.3.0/gems/bootsnap-1.1.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:17:in `require'
  | # ./vendor/bundle/ruby/2.3.0/gems/bootsnap-1.1.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:17:in `require'
  | # ./vendor/bundle/ruby/2.3.0/gems/bootsnap-1.1.6/lib/bootsnap/load_path_cache/core_ext/active_support.rb:43:in `load_missing_constant'
  | # ./spec/documentation/guide/content/structures/purchase_flow/shopping_cart/added_confirmation_modal_spec.rb:8:in `block (2 levels) in <top (required)>'
  | ...
```
I did not see issues like that with 1.1.5, but since they're random, maybe I've just been lucky. (I'm happy to test further if needed) I'm not sure if the lack of `.rb` at the end of the `LoadError` message is meaningful. (the file is present with the `.rb` extension`)

The errors happen at random with different files in different tests. I'm happy to provide more debugging details as needed.

The config is: load_path_cache true, autoload_paths_cache true, everything else false. It's not running in dev mode.

**Comment by viraptor** [comment]
Is there any more specific information I can provide to make it easier to debug?

**Comment by Envek** [comment]
I saw similar errors when running rspec in parallel or when launch development server while non-parallel rspec is running specs.

Errors was as follows:

```  
Errno::ENOTEMPTY:
  Directory not empty @ dir_s_rmdir - /path/to/project/tmp/cache/bootsnap-compile-cache
```

The cause was in RSpec configuration:

```ruby
config.after do
  Rails.cache.clear
end
```

So another process created cache entries while another tried to clear it.

I removed this RSpec config, but another option is to move bootsnap cache into other location.
